src/crimson/osd/osd.cc

```diff
@@ -269,7 +269,7 @@ seastar::future<> OSD::_write_superblock(
 	  meta_coll.create(t);
 	  meta_coll.store_superblock(t, superblock);
 	  logger().debug("OSD::_write_superblock: do_transaction...");
-	  return store.do_transaction(
+	  return store.get_local_shard_store().do_transaction(
```

Comment:
Worth a TODO:
// TODO: encapsulate into OSDMeta to use its FuturizedShardStore

Or fix it directly in this PR if convenient.